apps/partners-reference/lib/formatApplicationData.ts

```diff
@@ -0,0 +1,173 @@
+  This function eliminates those fields and parse to a proper format.
+*/
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+
+function getAddress(condition, addressData) {
+  const blankAddress: Address = {
+    id: undefined,
+    createdAt: undefined,
+    updatedAt: undefined,
+    street: "",
+    street2: "",
+    city: "",
+    state: "",
+    zipCode: "",
+  }
+
+  return condition ? addressData : blankAddress
+}
+
+export const formatApplicationData = (data: any, listingId, editMode: boolean) => {
+  const language: Language = data.application.language ? data.application.language : null
+
+  const submissionDate: Date | null = (() => {
+    // rename default (wrong property names)
+    const { birthDay: submissionDay, birthMonth: submissionMonth, birthYear: submissionYear } =
+      data.dateSubmitted || {}
+    const { hours, minutes = 0, seconds = 0, time } = data.timeSubmitted || {}
+
+    if (!submissionDay || !submissionMonth || !submissionYear) return null
+
+    const date = new Date()
+
+    date.setUTCDate(submissionDay)
+    date.setUTCMonth(submissionMonth - 1)
+    date.setUTCFullYear(submissionYear)
+
+    if (hours && minutes && seconds && time) {
+      const hourNumber = parseInt(hours, 10)
+      const hour24Clock = time === "am" ? hourNumber : hourNumber + 12
+      date.setUTCHours(hour24Clock, minutes, seconds)
+    } else {
+      date.setUTCHours(0, 0, 0, 0)
+    }
+
+    return date
+  })()
+
+  // create applicant
+  const applicant = ((): ApplicantUpdate => {
+    const { phoneNumber } = data
+    const { applicant: applicantData } = data.application
+    const phoneNumberType = applicantData.phoneNumberType ? applicantData.phoneNumberType : null
+    const noEmail = !applicantData.emailAddress
+    const noPhone = !phoneNumber
+    const workInRegion = applicantData?.workInRegion ? applicantData?.workInRegion : null
+    const emailAddress = applicantData?.emailAddress ? applicantData?.emailAddress : null
+
+    const workAddress = getAddress(
+      applicantData?.workInRegion === "yes",
+      applicantData?.workAddress
+    )
+
+    return {
+      ...applicantData,
+      ...data.dateOfBirth,
+      emailAddress,
+      workInRegion,
+      workAddress,
+      phoneNumber,
+      phoneNumberType,
+      noEmail,
+      noPhone,
+    }
+  })()
+
+  // additional phone
+  const {
+    additionalPhoneNumber: additionalPhoneNumberData,
+    additionalPhoneNumberType: additionalPhoneNumberTypeData,
+    mailingAddress: mailingAddressData,
+    additionalPhoneNumber,
+    contactPreferences,
+    sendMailToMailingAddress,
+    alternateContact,
+    accessibility,
+    preferences,
+    demographics,
+    preferredUnit,
+  } = data.application
+
+  const additionalPhone = !additionalPhoneNumberData
+  const additionalPhoneNumberType = additionalPhoneNumberTypeData
+    ? additionalPhoneNumberTypeData
+    : null
+
+  const mailingAddress = getAddress(sendMailToMailingAddress, mailingAddressData)
```

Comment:
What's the logic behind this? If `sendMailToMailingAddress` is falsey mailingAddress should be blank? Aren't we losing information then?

Reply:
Nope, in a single application, we have 3 types of addresses (residence, mailing, work) - so if a user doesn't select "checkbox", it means that we should store residence only (user doesn't see mailing address fields).